src/Nethermind/Nethermind.Mev/MevPlugin.cs

```diff
@@ -74,8 +76,8 @@ private BundlePool BundlePool
                 if (_bundlePool is null)
                 {
                     var (getFromApi, _) = _nethermindApi!.ForProducer;
-                    TxBundleSimulator txBundleSimulator = new(TracerFactory, getFromApi.GasLimitCalculator, getFromApi.Timestamper);
-                    _bundlePool = new BundlePool(getFromApi.BlockTree!, txBundleSimulator, getFromApi.FinalizationManager);
+                    TxBundleSimulator txBundleSimulator = new(TracerFactory, getFromApi.GasLimitCalculator, getFromApi.Timestamper, getFromApi.TxPool!, getFromApi.FinalizationManager!);
+                    _bundlePool = new BundlePool(getFromApi.BlockTree!, txBundleSimulator, getFromApi.FinalizationManager, getFromApi.Timestamper, _mevConfig, getFromApi.LogManager);
```

Comment:
when reorganizing we never produce new blocks really (think about it) so the concept of finalization in Bundle Pool seems to make no sense

Reply:
The idea was to have better bundle eviction based on finalization, especially useful after merge. But could be overkill.